main/src/ca/uwaterloo/flix/api/lsp/provider/completion/Completer.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2023 Lukas Rønn
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.api.lsp.provider.completion
+
+import ca.uwaterloo.flix.api.lsp.{CompletionItem, Index}
+import ca.uwaterloo.flix.language.ast.TypedAst
+
+/**
+  * A common super-type for completers.
+  */
+trait Completer {
+  /**
+    * Returns a List of LSP completion items for completer.
+    */
+  def getCompletions(implicit context: CompletionContext, index: Index, root: TypedAst.Root): Iterable[CompletionItem]
```

Comment:
But this should return `Completion`, not `CompletionItem`. In `CompletionProvider` we can map over each and translate them to completion items.

Reply:
This will later allow us to rank each completion differently.

Reply:
The high level point is that `Completion` is the data type ___we will use___ for ranking.

`CompletionItem` is a language server protocol data type that we cannot change the shape of.

Reply:
Noted! 

Brainfart from my point -> Rolled back to an earlier version, and forgot we changed it (Comment from earlier in the commit-history)...